x/configuration/client/cli/tx.go

```diff
@@ -80,151 +80,29 @@ func getCmdUpdateFees(cdc *codec.Codec) *cobra.Command {
-			}
-			if validURI != defaultRegex {
-				config.ValidURI = validURI
-			}
-			validResource, err := cmd.Flags().GetString("valid-resource")
-			if err != nil {
-				return err
-			}
-			if validResource != defaultRegex {
-				config.ValidResource = validResource
-			}
-			domainRenew, err := cmd.Flags().GetDuration("domain-renew-period")
-			if err != nil {
-				return err
-			}
-			if domainRenew != defaultDuration {
-				config.DomainRenewalPeriod = domainRenew
-			}
-			domainRenewCountMax, err := cmd.Flags().GetUint32("domain-renew-count-max")
-			if err != nil {
-				return err
-			}
-			if domainRenewCountMax != defaultNumber {
-				config.DomainRenewalCountMax = domainRenewCountMax
-			}
-			domainGracePeriod, err := cmd.Flags().GetDuration("domain-grace-period")
-			if err != nil {
-				return err
-			}
-			if domainGracePeriod != defaultNumber {
-				config.DomainGracePeriod = domainGracePeriod
-			}
-			accountRenewPeriod, err := cmd.Flags().GetDuration("account-renew-period")
-			if err != nil {
-				return err
-			}
-			if accountRenewPeriod != defaultNumber {
-				config.AccountRenewalPeriod = accountRenewPeriod
-			}
-			accountRenewCountMax, err := cmd.Flags().GetUint32("account-renew-count-max")
-			if err != nil {
-				return err
-			}
-			if accountRenewCountMax != defaultNumber {
-				config.AccountRenewalCountMax = accountRenewCountMax
-			}
-			accountGracePeriod, err := cmd.Flags().GetDuration("account-grace-period")
-			if err != nil {
-				return err
-			}
-			if accountGracePeriod != defaultDuration {
-				config.AccountGracePeriod = accountGracePeriod
-			}
-			resourceMax, err := cmd.Flags().GetUint32("resource-max")
-			if err != nil {
-				return err
-			}
-			if resourceMax != defaultNumber {
-				config.ResourcesMax = resourceMax
-			}
-			certificateSizeMax, err := cmd.Flags().GetUint64("certificate-size-max")
+
+			configFile, err := cmd.Flags().GetString("config-file")
 			if err != nil {
 				return err
 			}
-			if certificateSizeMax != defaultNumber {
-				config.CertificateSizeMax = certificateSizeMax
-			}
-			certificateCountMax, err := cmd.Flags().GetUint32("certificate-count-max")
+			f, err := os.Open(configFile)
 			if err != nil {
-				return err
-			}
-			if certificateCountMax != defaultNumber {
-				config.CertificateCountMax = certificateCountMax
+				return fmt.Errorf("unable to open config file: %s", err)
 			}
-			metadataSizeMax, err := cmd.Flags().GetUint64("metadata-size-max")
+			defer f.Close()
+			newConfig := new(types.Config)
+			err = json.NewDecoder(f).Decode(newConfig)
 			if err != nil {
-				return err
+				return fmt.Errorf("unable to decode json: %s", err)
 			}
-			if metadataSizeMax != defaultNumber {
-				config.MetadataSizeMax = metadataSizeMax
-			}
-
-			if err := config.Validate(); err != nil {
-				return err
-			}
-			// build msg
-			msg := &types.MsgUpdateConfig{
-				Signer:           signer,
-				NewConfiguration: *config,
+			msg := types.MsgUpdateConfig{
```

Comment:
We need some input validation just before this point, please.  eg, all config keys should be present and have seemingly valid values like periods should be at least a billion since they're in nanoseconds and regexes should compile.

Reply:
Msg validation is run here https://github.com/iov-one/iovns/pull/257/files#diff-7defa5dfa7355084a777be60745ac691R108

Reply:
Lets put together a list of requirements for configuration values to validate against. It would be cleaner